src/app/wallets/pay-on-chain.ts

```diff
@@ -310,37 +312,38 @@ const executePaymentViaOnChain = async ({
           return txHash
         }
 
-        let onChainTxFee = await onChainService.lookupOnChainFee({
+        let totalFee: Satoshis
+
+        const minerFee = await onChainService.lookupOnChainFee({
           txHash,
           scanDepth: ONCHAIN_SCAN_DEPTH_OUTGOING,
         })
 
-        if (onChainTxFee instanceof Error) {
-          logger.fatal({ err: onChainTxFee }, "impossible to get fee for onchain payment")
-          onChainTxFee = toSats(0)
+        if (minerFee instanceof Error) {
+          logger.error({ err: minerFee }, "impossible to get fee for onchain payment")
+          totalFee = estimatedFee
+        } else {
+          totalFee = withdrawFeeCalculator.onChainWithdrawalFee({
+            minerFee,
+            bankFee: toSats(senderAccount.withdrawFee),
+          })
         }
 
-        const fee = onChainTxFee
-          ? withdrawalFeeCalculator.onChainWithdrawalFee({
-              onChainFee: onChainTxFee,
-              walletFee: toSats(senderAccount.withdrawFee),
-            })
-          : estimatedFee
-        const sats = toSats(amountToSend + fee)
-        const usd = sats * usdPerSat
-        const usdFee = fee * usdPerSat
+        const sats = toSats(amountToSend + totalFee)
+        const usdDisplay = (sats * usdPerSat) as FiatAmount
```

Comment:
this need a helper similar to `toSats`, please avoid direct cast in use cases

Reply:
fixed